sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample02_EventHubsClients.md

```diff
@@ -0,0 +1,218 @@
+```
+The connection options are populated by default; you may set just the desired properties rather than creating a new instance, if you prefer.
+
+```C# Snippet:EventHubs_Sample02_ProducerTransportProperty
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producerOptions = new EventHubProducerClientOptions();
+producerOptions.ConnectionOptions.TransportType = EventHubsTransportType.AmqpWebSockets;
+
+var producer = new EventHubProducerClient(
+    connectionString,
+    eventHubName,
+    producerOptions);
+```
+
+### Setting a custom proxy
+
+A common scenario for adjusting the connection options is configuring a proxy.  Proxy support takes the form of the  [IWebProxy](https://docs.microsoft.com/dotnet/api/system.net.iwebproxy?view=netcore-3.1) interface, of which [WebProxy](https://docs.microsoft.com/dotnet/api/system.net.webproxy?view=netcore-3.1) is the most common default implementation.  Event Hubs supports a proxy only when using `AmqpWebSockets` as the transport type.
+
+For illustration, the `EventHubProducerClientOptions` are demonstrated, but the concept and form are common across the client options types.
+
+```C# Snippet:EventHubs_Sample02_ProducerProxyFullConnectionOptions
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producerOptions = new EventHubProducerClientOptions
+{
+    ConnectionOptions = new EventHubConnectionOptions
+    {
+        TransportType = EventHubsTransportType.AmqpWebSockets,
+        Proxy = new WebProxy("https://proxyserver:80", true)
+    }
+};
+
+var producer = new EventHubProducerClient(
+    connectionString,
+    eventHubName,
+    producerOptions);
+```
+
+The connection options are populated by default; you may set just the desired properties rather than creating a new instance, if you prefer.
+
+```C# Snippet:EventHubs_Sample02_ProducerProxyProperty
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var producerOptions = new EventHubProducerClientOptions();
+producerOptions.ConnectionOptions.TransportType = EventHubsTransportType.AmqpWebSockets;
+producerOptions.ConnectionOptions.Proxy = new WebProxy("https://proxyserver:80", true);
+
+var producer = new EventHubProducerClient(
+    connectionString,
+    eventHubName,
+    producerOptions);
+```
+
+### Using the default system proxy
+
+To use the default proxy for your environment, the recommended approach is to make use of  [HttpClient.DefaultProxy](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient.defaultproxy?view=netcore-3.1), which will attempt to detect proxy settings from the ambient environment in a manner consistent with expectations for the target platform.  Unfortunately, this member was added for .NET Core 3.1 and is not supported for earlier target frameworks.
```

Comment:
```suggestion
To use the default proxy for your environment, the recommended approach is to make use of [HttpClient.DefaultProxy](https://docs.microsoft.com/dotnet/api/system.net.http.httpclient.defaultproxy?view=netcore-3.1), which will attempt to detect proxy settings from the ambient environment in a manner consistent with expectations for the target platform.  Unfortunately, this member was added for .NET Core 3.1 and is not supported for earlier target frameworks.
```

Reply:
Applying manually to avoid throwing off my rebase.